src/Faithlife.Build/DotNetClassicTool.cs

```diff
@@ -60,7 +61,13 @@ public sealed class DotNetClassicTool
 
 		var packagesPath = Environment.GetEnvironmentVariable("NUGET_PACKAGES") ??
 			Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".nuget", "packages");
-		var packagePath = Path.Combine(packagesPath, packageName.ToLowerInvariant(), packageVersion);
+
+		// Find a matching package. If using a specific version, there will be exactly one match.
+		// If using floating versions, there may be multiple matches, so take the latest semantic version.
+		var packagePath = Directory.GetDirectories(Path.Combine(packagesPath, packageName.ToLowerInvariant()), packageVersion)
+			.OrderByDescending(fullPath => NuGetVersion.Parse(Path.GetFileName(fullPath)))
+			.FirstOrDefault();
+
 		if (!Directory.Exists(packagePath))
 			throw new BuildException($"Missing restored NuGet package: {packagePath}");
```

Comment:
Sorry I missed this before, but `packagePath` will now always be null here. Better to change the conditional to check for null and update the error message to include relevant details.